### ISSUE TITLE
[karmada init] prepare karmada failed.failed download file

### ISSUE THREAD

**Issue by RainbowMango** [karmada-io/karmada#1340]
**What happened**:
When installing Karmada by the following command:
```
kubectl karmada init
```

Getting the following error:
```
:I0131 10:46:44.646973   57124 deploy.go:201] download crds file name: /etc/karmada/crds.tar.gz
Error: prepare karmada failed.failed download file. url: https://github.com/karmada-io/karmada/releases/download/v1.0.0-43-g2060616c/crds.tar.gz code: 404
```

**What you expected to happen**:
From the error message we can tell that the version of `kubectl-karmada` is `v1.0.0-43-g2060616c`, not the released version but from built from the master branch. 

The version of `kubectl-karmada` should be shifted when building the URL, such as `v1.0.0-43-xxx` should be shifted to `v1.0.0`.
We can make a simple rule for this:
The version with the format of `vx.y.z-mm-nn` could be shifted to `vx.y.0` as no API change happens in the patched release.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`): v1.0.0-43-g2060616c
- Others:

cc @prodanlabs

**Comment by prodanlabs** [comment]
I have solved this problem, the PR I submitted before has not been merged, because of the conflict, it has not been submitted yet.

**Comment by RainbowMango** [comment]
Which PR is blocking? #1302 or #1337 ?

This bug fix should be picked to `release-1.0`, so can you help file a PR to fix this issue?
